app/models/prison.rb

```diff
@@ -1,32 +1,77 @@
 class Prison
-  DEFAULT_LEAD_DAYS = 3.freeze
-  DEFAULT_BOOKING_WINDOW = 28.freeze
+  include Virtus.model
+
+  DEFAULT_ADULT_AGE =       18
+  DEFAULT_BOOKING_WINDOW =  28
+  DEFAULT_LEAD_DAYS =       3
+
+  attribute :address
+  attribute :adult_age, Integer,
+    default: DEFAULT_ADULT_AGE,
+    lazy: true
+  attribute :booking_window, Integer,
+    default: DEFAULT_BOOKING_WINDOW,
+    lazy: true
+  attribute :canned_responses, Boolean
+  attribute :email, String
+  attribute :enabled, Boolean
+  attribute :finder_slug, String
+  attribute :lead_days, Integer,
+    default: DEFAULT_LEAD_DAYS,
+    lazy: true
+  attribute :name, String
+  attribute :nomis_id, String
+  attribute :phone, String
+  attribute :reason
+  attribute :slot_anomalies
+  attribute :slots
+  attribute :unbookable, Boolean
+  attribute :works_weekends, Boolean, default: false, lazy: true
 
   class PrisonNotFound < StandardError; end
 
-  def self.find(prison_name)
-    prison_hash = Rails.configuration.prison_data[prison_name]
-    if prison_hash.nil?
-      raise PrisonNotFound, "Can't find prison #{prison_name.inspect}"
+  def self.find(name_or_nomis, raise_error = false)
+    prison = all.detect { |p|
+      p.name == name_or_nomis || p.nomis_id == name_or_nomis
+    }
+    if raise_error && prison.nil?
+      raise PrisonNotFound, "Can't find prison #{name_or_nomis.inspect}"
     end
-    new(prison_name, prison_hash)
+    prison
```

Comment:
@dan-palmer duplicating the found prison breaks a fair number of specs at this point.  I'm going to leave it for now and raise a ticket to look at it shortly.